src/hotspot/share/classfile/classLoaderExt.cpp

```diff
@@ -81,20 +81,35 @@ void ClassLoaderExt::setup_app_search_path(JavaThread* current) {
 
 void ClassLoaderExt::process_module_table(JavaThread* current, ModuleEntryTable* met) {
   ResourceMark rm(current);
-  class Process : public ModuleClosure {
+  GrowableArray<char*>* module_paths = new GrowableArray<char*>(5);
+
+  class ModulePathsGatherer : public ModuleClosure {
     JavaThread* _current;
+    int _total_count;
+    GrowableArray<char*>* _module_paths;
    public:
-    Process(JavaThread* current) : _current(current) {}
+    ModulePathsGatherer(JavaThread* current, GrowableArray<char*>* module_paths) :
+      _current(current), _total_count(0), _module_paths(module_paths) {}
     void do_module(ModuleEntry* m) {
       char* path = m->location()->as_C_string();
       if (strncmp(path, "file:", 5) == 0) {
         path = ClassLoader::skip_uri_protocol(path);
-        ClassLoader::setup_module_search_path(_current, path);
+        char* path_copy = NEW_RESOURCE_ARRAY(char, strlen(path) + 1);
+        strcpy(path_copy, path);
+        _module_paths->at_put_grow(_total_count++, path_copy);
```

Comment:
You can use `_module_paths->append()` to avoid keeping the `_total_count`

Reply:
Fixed.
Thanks for your review.